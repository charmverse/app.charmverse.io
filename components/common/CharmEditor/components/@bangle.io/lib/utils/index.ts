export * from './hooks';
export * from './safe-js-callbacks';
export * from './utility';

