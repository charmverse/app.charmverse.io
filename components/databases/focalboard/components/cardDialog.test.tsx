// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import '@testing-library/jest-dom'
import {act, render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import React from 'react'
import {Provider as ReduxProvider} from 'react-redux'
import {mocked} from 'ts-jest/utils'

import mutator from '../mutator'
import {Utils} from '../utils'
import {TestBlockFactory} from '../test/testBlockFactory'
import {mockDOM, mockStateStore, wrapDNDIntl} from '../testUtils'

import CardDialog from './cardDialog'

jest.mock('../mutator')
jest.mock('../utils')
jest.mock('draft-js/lib/generateRandomKey', () => () => '123')

const mockedUtils = mocked(Utils, true)
const mockedMutator = mocked(mutator, true)
mockedUtils.createGuid.mockReturnValue('test-id')

beforeAll(() => {
    mockDOM()
})
describe('components/cardDialog', () => {
    const board = TestBlockFactory.createBoard()
    board.fields.cardProperties = []
    board.id = 'test-id'
    board.rootId = board.id
    const boardView = TestBlockFactory.createBoardView(board)
    boardView.id = board.id
    const card = TestBlockFactory.createCard(board)
    card.id = board.id
    card.createdBy = 'user-id-1'

    const state = {
        clientConfig: {
            value: {
                featureFlags: {
                    subscriptions: true,
                },
            },
        },
        comments: {
            comments: {},
        },
        contents: {},
        cards: {
            cards: {
                [card.id]: card,
            },
        },
        users: {
            workspaceUsers: {
                1: {username: 'abc'},
                2: {username: 'd'},
                3: {username: 'e'},
                4: {username: 'f'},
                5: {username: 'g'},
            },
            blockSubscriptions: [],
        },
    }
    const store = mockStateStore([], state)
    beforeEach(() => {
        jest.clearAllMocks()
    })
    test('should match snapshot', async () => {
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
            container = result.container
        })
        expect(container).toMatchSnapshot()
    })
    test('return a cardDialog readonly', async () => {
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={true}
                    />
                </ReduxProvider>,
            ))
            container = result.container
        })
        expect(container).toMatchSnapshot()
    })
    test('return cardDialog and do a close action', async () => {
        const closeFn = jest.fn()
        await act(async () => {
            render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={closeFn}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
        })
        const buttonElement = screen.getByRole('button', {name: 'Close dialog'})
        userEvent.click(buttonElement)
        expect(closeFn).toBeCalledTimes(1)
    })
    test('return cardDialog menu content', async () => {
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
            container = result.container
        })
        const buttonMenu = screen.getAllByRole('button', {name: 'menuwrapper'})[0]
        userEvent.click(buttonMenu)
        expect(container).toMatchSnapshot()
    })
    test('return cardDialog menu content and verify delete action', async () => {
        await act(async () => {
            render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
        })
        const buttonMenu = screen.getAllByRole('button', {name: 'menuwrapper'})[0]
        userEvent.click(buttonMenu)
        const buttonDelete = screen.getByRole('button', {name: 'Delete'})
        userEvent.click(buttonDelete)

        const confirmDialog = screen.getByTitle('Confirmation Dialog Box')
        expect(confirmDialog).toBeDefined()

        const confirmButton = screen.getByTitle('Delete')
        expect(confirmButton).toBeDefined()

        //click delete button
        userEvent.click(confirmButton!)

        // should be called once on confirming delete
        expect(mockedMutator.deleteBlock).toBeCalledTimes(1)
    })

    test('return cardDialog menu content and cancel delete confirmation do nothing', async () => {
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
            container = result.container
        })

        const buttonMenu = screen.getAllByRole('button', {name: 'menuwrapper'})[0]
        userEvent.click(buttonMenu)
        const buttonDelete = screen.getByRole('button', {name: 'Delete'})
        userEvent.click(buttonDelete)

        const confirmDialog = screen.getByTitle('Confirmation Dialog Box')
        expect(confirmDialog).toBeDefined()

        const cancelButton = screen.getByTitle('Cancel')
        expect(cancelButton).toBeDefined()

        //click delete button
        userEvent.click(cancelButton!)

        // should do nothing  on cancel delete dialog
        expect(container).toMatchSnapshot()
    })

    test('return cardDialog menu content and do a New template from card', async () => {
        await act(async () => {
            render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
        })
        const buttonMenu = screen.getAllByRole('button', {name: 'menuwrapper'})[0]
        userEvent.click(buttonMenu)
        const buttonTemplate = screen.getByRole('button', {name: 'New template from card'})
        userEvent.click(buttonTemplate)
        expect(mockedMutator.duplicateCard).toBeCalledTimes(1)
    })

    test('return cardDialog menu content and do a copy Link', async () => {
        await act(async () => {
            render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
        })
        const buttonMenu = screen.getAllByRole('button', {name: 'menuwrapper'})[0]
        userEvent.click(buttonMenu)
        const buttonCopy = screen.getByRole('button', {name: 'Copy link'})
        userEvent.click(buttonCopy)
        expect(mockedUtils.copyTextToClipboard).toBeCalledTimes(1)
    })

    test('already following card', async () => {
        // simply doing {...state} gives a TypeScript error
        // when you try updating it's values.
        const newState = JSON.parse(JSON.stringify(state))
        newState.users.blockSubscriptions = [{blockId: card.id}]
        newState.clientConfig = {
            value: {
                featureFlags: {
                    subscriptions: true,
                },
            },
        }

        const newStore = mockStateStore([], newState)

        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ReduxProvider store={newStore}>
                    <CardDialog
                        board={board}
                        activeView={boardView}
                        views={[boardView]}
                        cards={[card]}
                        cardId={card.id}
                        onClose={jest.fn()}
                        showCard={jest.fn()}
                        readonly={false}
                    />
                </ReduxProvider>,
            ))
            container = result.container
        })
        expect(container).toMatchSnapshot()
    })
})
