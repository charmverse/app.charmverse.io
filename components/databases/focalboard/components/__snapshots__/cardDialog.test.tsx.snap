// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/cardDialog already following card 1`] = `
<div>
  <div
    class="Dialog dialog-back undefined"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <button
            aria-label="Close dialog"
            class="Button IconButton IconButton--large"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
          <div
            class="cardToolbar"
          >
            <button
              type="button"
            >
              <span>
                Following
              </span>
            </button>
          </div>
          <div
            aria-label="menuwrapper"
            class="MenuWrapper"
            role="button"
          >
            <button
              class="Button IconButton IconButton--large"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
        </div>
        <div
          class="CardDetail content"
        >
          <div
            class="BlockIconSelector"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <div
                class="octo-icon size-l"
              >
                <span>
                  i
                </span>
              </div>
            </div>
          </div>
          <div
            class="EditableAreaWrap"
          >
            <textarea
              class="EditableArea Editable title"
              height="0"
              placeholder="Untitled"
              rows="1"
              spellcheck="true"
              title="title"
            >
              title
            </textarea>
            <div
              class="EditableAreaContainer"
            >
              <textarea
                aria-hidden="true"
                class="EditableAreaReference Editable title"
                dir="auto"
                disabled=""
                rows="1"
              >
                title
              </textarea>
            </div>
          </div>
          <div
            class="octo-propertylist CardDetailProperties"
          >
            <div
              class="octo-propertyname add-property"
            >
              <div
                aria-label="menuwrapper"
                class="MenuWrapper"
                role="button"
              >
                <button
                  type="button"
                >
                  <span>
                    + Add a property
                  </span>
                </button>
              </div>
            </div>
          </div>
          <hr />
          <div
            class="CommentsList"
          >
            <div
              class="CommentsList__new"
            >
              <img
                class="comment-avatar"
              />
              <div
                class="MarkdownEditor octo-editor newcomment "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              
            </div>
            <hr
              class="CommentsList__divider"
            />
          </div>
        </div>
        <div
          class="CardDetail content fullwidth content-blocks"
        >
          <div
            class="octo-content CardDetailContents"
          >
            <div
              class="octo-block"
            >
              <div
                class="octo-block-margin"
              />
              <div
                class="MarkdownEditor octo-editor  "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="CardDetailContentsMenu content add-content"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <button
                type="button"
              >
                <span>
                  Add content
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/cardDialog return a cardDialog readonly 1`] = `
<div>
  <div
    class="Dialog dialog-back undefined"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <button
            aria-label="Close dialog"
            class="Button IconButton IconButton--large"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
          <div
            class="cardToolbar"
          >
            <button
              type="button"
            >
              <span>
                Follow
              </span>
            </button>
          </div>
        </div>
        <div
          class="CardDetail content"
        >
          <div
            class="BlockIconSelector"
          >
            <div
              class="octo-icon size-l readonly"
            >
              <span>
                i
              </span>
            </div>
          </div>
          <div
            class="EditableAreaWrap"
          >
            <textarea
              class="EditableArea Editable readonly title"
              height="0"
              placeholder="Untitled"
              readonly=""
              rows="1"
              spellcheck="true"
              title="title"
            >
              title
            </textarea>
            <div
              class="EditableAreaContainer"
            >
              <textarea
                aria-hidden="true"
                class="EditableAreaReference Editable readonly title"
                dir="auto"
                disabled=""
                rows="1"
              >
                title
              </textarea>
            </div>
          </div>
          <div
            class="octo-propertylist CardDetailProperties"
          />
          <hr />
          <div
            class="CommentsList"
          />
        </div>
        <div
          class="CardDetail content fullwidth content-blocks"
        >
          <div
            class="octo-content CardDetailContents"
          >
            <div
              class="octo-block"
            >
              <div
                class="octo-block-margin"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/cardDialog return cardDialog menu content 1`] = `
<div>
  <div
    class="Dialog dialog-back undefined"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <button
            aria-label="Close dialog"
            class="Button IconButton IconButton--large"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
          <div
            class="cardToolbar"
          >
            <button
              type="button"
            >
              <span>
                Follow
              </span>
            </button>
          </div>
          <div
            aria-label="menuwrapper"
            class="MenuWrapper"
            role="button"
          >
            <button
              class="Button IconButton IconButton--large"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
            <div
              class="Menu noselect left"
            >
              <div
                class="menu-contents"
              >
                <div
                  class="menu-options"
                >
                  <div
                    aria-label="Delete"
                    class="MenuOption TextOption menu-option"
                    role="button"
                  >
                    <i
                      class="CompassIcon icon-trash-can-outline trash-can-outline"
                    />
                    <div
                      class="menu-name"
                    >
                      Delete
                    </div>
                    <div
                      class="noicon"
                    />
                  </div>
                  <div
                    aria-label="Copy link"
                    class="MenuOption TextOption menu-option"
                    role="button"
                  >
                    <i
                      class="CompassIcon icon-link-variant LinkIcon"
                    />
                    <div
                      class="menu-name"
                    >
                      Copy link
                    </div>
                    <div
                      class="noicon"
                    />
                  </div>
                  <div
                    aria-label="New template from card"
                    class="MenuOption TextOption menu-option"
                    role="button"
                  >
                    <div
                      class="noicon"
                    />
                    <div
                      class="menu-name"
                    >
                      New template from card
                    </div>
                    <div
                      class="noicon"
                    />
                  </div>
                </div>
                <div
                  class="menu-spacer hideOnWidescreen"
                />
                <div
                  class="menu-options hideOnWidescreen"
                >
                  <div
                    aria-label="Cancel"
                    class="MenuOption TextOption menu-option menu-cancel"
                    role="button"
                  >
                    <div
                      class="noicon"
                    />
                    <div
                      class="menu-name"
                    >
                      Cancel
                    </div>
                    <div
                      class="noicon"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="CardDetail content"
        >
          <div
            class="BlockIconSelector"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <div
                class="octo-icon size-l"
              >
                <span>
                  i
                </span>
              </div>
            </div>
          </div>
          <div
            class="EditableAreaWrap"
          >
            <textarea
              class="EditableArea Editable title"
              height="0"
              placeholder="Untitled"
              rows="1"
              spellcheck="true"
              title="title"
            >
              title
            </textarea>
            <div
              class="EditableAreaContainer"
            >
              <textarea
                aria-hidden="true"
                class="EditableAreaReference Editable title"
                dir="auto"
                disabled=""
                rows="1"
              >
                title
              </textarea>
            </div>
          </div>
          <div
            class="octo-propertylist CardDetailProperties"
          >
            <div
              class="octo-propertyname add-property"
            >
              <div
                aria-label="menuwrapper"
                class="MenuWrapper"
                role="button"
              >
                <button
                  type="button"
                >
                  <span>
                    + Add a property
                  </span>
                </button>
              </div>
            </div>
          </div>
          <hr />
          <div
            class="CommentsList"
          >
            <div
              class="CommentsList__new"
            >
              <img
                class="comment-avatar"
              />
              <div
                class="MarkdownEditor octo-editor newcomment "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              
            </div>
            <hr
              class="CommentsList__divider"
            />
          </div>
        </div>
        <div
          class="CardDetail content fullwidth content-blocks"
        >
          <div
            class="octo-content CardDetailContents"
          >
            <div
              class="octo-block"
            >
              <div
                class="octo-block-margin"
              />
              <div
                class="MarkdownEditor octo-editor  "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="CardDetailContentsMenu content add-content"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <button
                type="button"
              >
                <span>
                  Add content
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/cardDialog return cardDialog menu content and cancel delete confirmation do nothing 1`] = `
<div>
  <div
    class="Dialog dialog-back undefined"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <button
            aria-label="Close dialog"
            class="Button IconButton IconButton--large"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
          <div
            class="cardToolbar"
          >
            <button
              type="button"
            >
              <span>
                Follow
              </span>
            </button>
          </div>
          <div
            aria-label="menuwrapper"
            class="MenuWrapper"
            role="button"
          >
            <button
              class="Button IconButton IconButton--large"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
        </div>
        <div
          class="CardDetail content"
        >
          <div
            class="BlockIconSelector"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <div
                class="octo-icon size-l"
              >
                <span>
                  i
                </span>
              </div>
            </div>
          </div>
          <div
            class="EditableAreaWrap"
          >
            <textarea
              class="EditableArea Editable title"
              height="0"
              placeholder="Untitled"
              rows="1"
              spellcheck="true"
              title="title"
            >
              title
            </textarea>
            <div
              class="EditableAreaContainer"
            >
              <textarea
                aria-hidden="true"
                class="EditableAreaReference Editable title"
                dir="auto"
                disabled=""
                rows="1"
              >
                title
              </textarea>
            </div>
          </div>
          <div
            class="octo-propertylist CardDetailProperties"
          >
            <div
              class="octo-propertyname add-property"
            >
              <div
                aria-label="menuwrapper"
                class="MenuWrapper"
                role="button"
              >
                <button
                  type="button"
                >
                  <span>
                    + Add a property
                  </span>
                </button>
              </div>
            </div>
          </div>
          <hr />
          <div
            class="CommentsList"
          >
            <div
              class="CommentsList__new"
            >
              <img
                class="comment-avatar"
              />
              <div
                class="MarkdownEditor octo-editor newcomment "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              
            </div>
            <hr
              class="CommentsList__divider"
            />
          </div>
        </div>
        <div
          class="CardDetail content fullwidth content-blocks"
        >
          <div
            class="octo-content CardDetailContents"
          >
            <div
              class="octo-block"
            >
              <div
                class="octo-block-margin"
              />
              <div
                class="MarkdownEditor octo-editor  "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="CardDetailContentsMenu content add-content"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <button
                type="button"
              >
                <span>
                  Add content
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/cardDialog should match snapshot 1`] = `
<div>
  <div
    class="Dialog dialog-back undefined"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <button
            aria-label="Close dialog"
            class="Button IconButton IconButton--large"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
          <div
            class="cardToolbar"
          >
            <button
              type="button"
            >
              <span>
                Follow
              </span>
            </button>
          </div>
          <div
            aria-label="menuwrapper"
            class="MenuWrapper"
            role="button"
          >
            <button
              class="Button IconButton IconButton--large"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
        </div>
        <div
          class="CardDetail content"
        >
          <div
            class="BlockIconSelector"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <div
                class="octo-icon size-l"
              >
                <span>
                  i
                </span>
              </div>
            </div>
          </div>
          <div
            class="EditableAreaWrap"
          >
            <textarea
              class="EditableArea Editable title"
              height="0"
              placeholder="Untitled"
              rows="1"
              spellcheck="true"
              title="title"
            >
              title
            </textarea>
            <div
              class="EditableAreaContainer"
            >
              <textarea
                aria-hidden="true"
                class="EditableAreaReference Editable title"
                dir="auto"
                disabled=""
                rows="1"
              >
                title
              </textarea>
            </div>
          </div>
          <div
            class="octo-propertylist CardDetailProperties"
          >
            <div
              class="octo-propertyname add-property"
            >
              <div
                aria-label="menuwrapper"
                class="MenuWrapper"
                role="button"
              >
                <button
                  type="button"
                >
                  <span>
                    + Add a property
                  </span>
                </button>
              </div>
            </div>
          </div>
          <hr />
          <div
            class="CommentsList"
          >
            <div
              class="CommentsList__new"
            >
              <img
                class="comment-avatar"
              />
              <div
                class="MarkdownEditor octo-editor newcomment "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              
            </div>
            <hr
              class="CommentsList__divider"
            />
          </div>
        </div>
        <div
          class="CardDetail content fullwidth content-blocks"
        >
          <div
            class="octo-content CardDetailContents"
          >
            <div
              class="octo-block"
            >
              <div
                class="octo-block-margin"
              />
              <div
                class="MarkdownEditor octo-editor  "
              >
                <div
                  class="octo-editor-preview octo-placeholder"
                  data-testid="preview-element"
                />
                <div
                  class="MarkdownEditorInput MarkdownEditorInput--IsNotEditing"
                >
                  <div
                    class="DraftEditor-root"
                  >
                    <div
                      class="DraftEditor-editorContainer"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-expanded="false"
                        class="notranslate public-DraftEditor-content"
                        contenteditable="true"
                        role="combobox"
                        spellcheck="false"
                        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                      >
                        <div
                          data-contents="true"
                        >
                          <div
                            class=""
                            data-block="true"
                            data-editor="123"
                            data-offset-key="123-0-0"
                          >
                            <div
                              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                              data-offset-key="123-0-0"
                            >
                              <span
                                data-offset-key="123-0-0"
                              >
                                <br
                                  data-text="true"
                                />
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="CardDetailContentsMenu content add-content"
          >
            <div
              aria-label="menuwrapper"
              class="MenuWrapper"
              role="button"
            >
              <button
                type="button"
              >
                <span>
                  Add content
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
